chat/models.tsp

```diff
@@ -204,149 +183,97 @@ model CreateChatCompletionResponse {
-  | "gpt-3.5-turbo"
-  | "gpt-3.5-turbo-16k"
-  | "gpt-3.5-turbo-0301"
-  | "gpt-3.5-turbo-0613"
-  | "gpt-3.5-turbo-1106"
-  | "gpt-3.5-turbo-0125"
-  | "gpt-3.5-turbo-16k-0613";
-
-@oneOf
-union Stop {
-  string,
-  StopSequences,
-}
-
-@minItems(1)
-@maxItems(4)
-model StopSequences is string[];
-
-/** Usage statistics for the completion request. */
-model CompletionUsage {
-  /** Number of tokens in the prompt. */
-  prompt_tokens: safeint;
-
-  /** Number of tokens in the generated completion */
-  completion_tokens: safeint;
-
-  /** Total number of tokens used in the request (prompt + completion). */
-  total_tokens: safeint;
-}
-
 model ChatCompletionTool {
-  /** The type of the tool. Currently, only `function` is supported. */
+  @doc("""
+  The type of the tool. Currently, only `function` is supported.
+  """)
   type: "function";
 
   function: FunctionObject;
 }
 
-/**
- * Controls which (if any) function is called by the model. `none` means the model will not call a
- * function and instead generates a message. `auto` means the model can pick between generating a
- * message or calling a function. Specifying a particular function via
- * `{"type": "function", "function": {"name": "my_function"}}` forces the model to call that
- * function.
- * 
- * `none` is the default when no functions are present. `auto` is the default if functions are
- * present.
- */
-@oneOf
-@extension("x-oaiExpandable", true)
-union ChatCompletionToolChoiceOption {
-  "none",
-  "auto",
-  ChatCompletionNamedToolChoice,
-}
-
 /** Specifies a tool the model should use. Use to force the model to call a specific function. */
 model ChatCompletionNamedToolChoice {
-  /** The type of the tool. Currently, only `function` is supported. */
+  @doc("""
+  The type of the tool. Currently, only `function` is supported.
+  """)
   type: "function";
 
   function: {
     /** The name of the function to call. */
     name: string;
-  }
+  };
 }
 
-@oneOf
-union ChatCompletionRequestUserMessageContent {
-  /** The text contents of the message. */
-  string,
+@doc("""
+Controls which (if any) function is called by the model.
+`none` means the model will not call a function and instead generates a message.
+`auto` means the model can pick between generating a message or calling a function.
+Specifying a particular function via `{"type": "function", "function": {"name": "my_function"}}` forces the model to call that function.
 
-  /**
-   * An array of content parts with a defined type, each can be of type `text` or `image_url` when
-   * passing in images. You can pass multiple images by adding multiple `image_url` content parts.
-   * Image input is only supported when using the `gpt-4-visual-preview` model.
-   */
-  ChatCompletionRequestMessageContentParts,
-};
+`none` is the default when no functions are present. `auto` is the default if functions are present.
+""")
+@extension("x-oaiExpandable", true)
+union ChatCompletionToolChoiceOption {
+"none" | "auto",
```

Comment:
Should it be a comma instead of a pipe between "none" and "auto":

```
"none",
"auto",
ChatCompletionNameToolChoice
```

Reply:
I believe they're equivalent:
https://typespec.io/playground?c=dW5pb24gRm9vIHsNCiAgImhlbGxvIiB8ICJ3b3JsZCINCn0NCg0KxidCYXLNJyzFDMkp&e=%40typespec%2Fopenapi3&options=%7B%7D

Reply:
(we can update this later if we don't like the format, though)

Reply:
As written, it's a more literal translation of the `oneOf` schema:
```
      oneOf:
        - type: string
          description: >
            `none` means the model will not call a function and instead generates a message.
            `auto` means the model can pick between generating a message or calling a function.
          enum: [none, auto]
        - $ref: "#/components/schemas/ChatCompletionNamedToolChoice"
```